accepted/2023/simplify-workload-versioning.md

```diff
@@ -0,0 +1,140 @@
+   Manifest Version:    8.0.0-preview.4.23181.9/8.0.100-preview.4
+   Manifest Path:       C:\git\dotnet-sdk-main\.dotnet\sdk-manifests\8.0.100-preview.4\microsoft.net.workload.mono.toolchain.current\WorkloadManifest.json
+   Install Type:        FileBased
+```
+
+When updating workloads, console output will include the old and new workload patch versions:
+
+```
+> dotnet workload install wasm-tools
+Checking for updated workloads version...
+Updating workloads version from 8.0.201.0 to 8.0.201.2...
+Installing workload manifest microsoft.net.sdk.android version 34.0.0-preview.4.230...
+<Further workload manifest update messages>
+Installing pack Microsoft.NET.Runtime.WebAssembly.Sdk version 8.0.0-preview.4.23181.9...
+<Further workload pack installation manifests>
+Garbage collecting for SDK feature band(s) 8.0.200...
+
+Successfully updated workloads version from 8.0.201.0 to 8.0.201.2.
+Successfully installed workload(s) wasm-tools.
+```
+
+The proposed [workload history](https://github.com/dotnet/sdk/pull/30486) command will also display the workload patch versions.
+
+The .NET SDK will periodically (once a day, by default) check for updated workload versions in order to notify users if there is an update available.  Currently, commands such as `dotnet build` print out the following message:
+
+```
+Workload updates are available. Run `dotnet workload list` for more information.
+```
+
+The `dotnet workload list` command, in turn, prints out the following message:
+
+```
+Updates are available for the following workload(s): wasm-tools. Run `dotnet workload update` to get the latest.
+```
+
+We will modify both commands to print the same message:
+
+```
+A workload update (version 8.0.201.2) is available. Run `dotnet workload update` to install the latest workloads.
+```
+
+We will add a new `--version` option to `dotnet workload update` to allow installing (or downgrading to) a specific workloads version:
+
+```
+> dotnet workload update --version 8.0.201.0
+Updating workloads version from 8.0.201.2 to 8.0.201.0...
+<Manifest and pack installation / garbage collection messages>
+Successfully updated workloads version from 8.0.201.2 to 8.0.201.0.
+```
+
+## Specifying workload versions with global.json
+
+We will add support for specifying the workloads version in global.json.  For example:
+
+```json
+{
+  "sdk": {
+    "version": "8.0.201.0",
+    "rollForward": "disable"
+  }
+}
+```
+
+This would force the SDK to use workloads version `8.0.201.0`, and would error if that version was not installed.
+
+We will support side-by-side workload version installations.  If 8.0.201.2 is installed, we would support running `dotnet workload install --version 8.0.201.0` to install that version of the workloads.  After installing the earlier version, the .NET SDK would still by default use the latest installed workloads version.  An earlier workloads version would only be used if it was specified in a global.json file.
```

Comment:
This would require having SxS manifests instead of requiring them to upgrade within a band across SDK patches? It would help with the downgrade functionality as well I think.